core/domain/learner_progress_domain_test.py

```diff
@@ -0,0 +1,76 @@
+# coding: utf-8
+#
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for learner progress domain objects."""
+
+from core.domain import learner_progress_domain
+from core.tests import test_utils
+
+
+class LearnerProgressUnitTests(test_utils.GenericTestBase):
+    """Tests the learner progress domain object."""
+
+    def test_initialization(self):
+        """Tests init method."""
+        user_learner_progress = (learner_progress_domain.LearnerProgress(
+            [], [], [], [], [], []))
+
+        self.assertEqual(
+            user_learner_progress.incomplete_exp_summaries, [])
+        self.assertEqual(
+            user_learner_progress.incomplete_collection_summaries, [])
+        self.assertEqual(user_learner_progress.completed_exp_summaries, [])
+        self.assertEqual(
+            user_learner_progress.completed_collection_summaries, [])
+        self.assertEqual(
+            user_learner_progress.exploration_playlist_summaries, [])
+        self.assertEqual(
+            user_learner_progress.collection_playlist_summaries, [])
+
+
+class ActivityIdsInLearnerDashboardUnitTests(test_utils.GenericTestBase):
+    """Tests the activity ids in learner dashboard domain object."""
+
+    def test_to_dict(self):
+        incomplete_exp_id = ['0']
```

Comment:
These should be named plurally since they represent lists of exp IDs.

Reply:
Done!